test/metabase/lib/query_test.cljc

```diff
@@ -95,6 +96,31 @@
 
               (lib/query meta/metadata-provider converted-query))))))
 
+(deftest ^:parallel converted-query-leaves-stage-metadata-refs-alone
+  (let [query (-> (lib/query meta/metadata-provider (meta/table-metadata :people))
+                  (lib/expression "BirthMonth" (lib/+ 1 1))
+                  (as-> $q (lib/breakout $q (m/find-first #(= (:name %) "BirthMonth") (lib/breakoutable-columns $q))))
+                  (lib/aggregate (lib/count))
+                  (lib/append-stage)
+                  (lib/aggregate (lib/count)))]
+    (is (=? {:stages [{:lib/stage-metadata {:columns [{:field-ref [:expression "BirthMonth" {:base-type :type/Integer}]} {}]}} {}]}
+          (lib/query meta/metadata-provider (assoc-in (lib.convert/->pMBQL (lib.convert/->legacy-MBQL query))
+                                                        [:stages 0 :lib/stage-metadata]
+                                                        {:columns [{:base-type :type/Float,
+                                                                    :display-name "BirthMonth",
+                                                                    :field-ref [:expression
+                                                                                "BirthMonth"
+                                                                                {:base-type :type/Integer}],
+                                                                    :name "BirthMonth",
+                                                                    :lib/type :metadata/column}
+                                                                   {:base-type :type/Integer,
+                                                                    :display-name "Count",
+                                                                    :field-ref [:aggregation 0],
+                                                                    :name "count",
+                                                                    :semantic-type :type/Quantity,
+                                                                    :lib/type :metadata/column}],
+                                                         :lib/type :metadata/results}))))))
```

Comment:
Why not name this map and check that it is completely preserved when running `lib.convert->pMBQL`?